### DISCUSSION TITLE
Set page size when converting images to pdf

### DISCUSSION THREAD

**Discussion by tomasr78** [dlemstra/Magick.NET#948] [Help]
The ImageMagick tutorial shows that it is possible to convert images to PDF and set page size by including page size name like A4

`convert -density 80 -page a4 input_A.jpg input_B.jpg output.pdf`

When I try to do the same(?) I get exception "Invalid geometry" at  `magickReadSettings.Page = new MagickGeometry("a4");`

Is it possible to set page size by name?

```
            var magickReadSettings = new MagickReadSettings
            {

               Compression = CompressionMethod.JPEG,
            };

            magickReadSettings.Page = new MagickGeometry("a4");
            
            using (var image = new MagickImage())
            {
                image.Read(@"test.heic", magickReadSettings);


                image.Write(@"output.pdf");
            }
```

**Comment by dlemstra** [comment]
I have changed your issue into a discussion. 

You should not set the page size with `magickReadSettings` but set `image.Page` after the image is read instead:

```C#
image.Page = MagickGeometry.FromPageSize("a4");
```

**Reply by tomasr78** [reply]
Thank you, it works. And to finish my solution I would like to know how to align an image in the page center. Now it aligns at the left of the page. 

Also, are there any parameters to set the margins of the page?

**Reply by dlemstra** [reply]
You will need probably need to extent the size of the image to the page size and when you extent the image you should use `Gravity.Center`.

**Comment by tomasr78** [comment]
Dirk, maybe you could help me why my code blurry images in PDF even when image size is bigger than page dimension. I mean images become blurry even when making them smaller and fitting to PDF page in the center. 

```
            using (var imageCollection = new MagickImageCollection())
            {
                imageCollection.Add(MagickImage(@"C:\myfile\jun_pdf_image.png"));
                imageCollection.Add(MagickImage(@"C:\myfile\original.jpg"));                                
                imageCollection.Write(@"C:\Projects\_temp\output.pdf");
            }

 private static MagickImage MagickImage(string imageFile)
        {
            var image = new MagickImage();
            var magickReadSettings = new MagickReadSettings();
            image.Read(imageFile, magickReadSettings);

            var fromPageSize = MagickGeometry.FromPageSize("a4");

            image.Resize(fromPageSize.Width, fromPageSize.Height);
            image.Extent(fromPageSize.Width, fromPageSize.Height, Gravity.Center);                                   

            image.Page = fromPageSize;

            return image;
        }


```
If I remove Resize and Extend methods when the images look sharp as original in PDF but then images are not centered in PDF. I am trying to center it that's why I must Resize and Extend. The Extend does not work without Resize and Resize produces a blurry image. 
Spends several hours to understand what's is causing blur in resized(downsized) images without any luck.

**Comment by dlemstra** [comment]
Could you start a new question and explain in more detail what is happening?

**Reply by tomasr78** [reply]
I have created new thread and tried to explain in more details, also attached example files https://github.com/dlemstra/Magick.NET/discussions/1034
